src/Autocomplete/src/EntityAutocompleterInterface.php

```diff
@@ -18,30 +18,52 @@
 /**
  * Interface for classes that will have an "autocomplete" endpoint exposed.
  *
+ * @template T of object
+ *
+ * @method array getResults(object $entity) Returns the autocomplete result, usually the label as "text" and the value as "value".
```

Comment:
Thinking out loud here, but this is dimishing the whole interface here. 

getLabel and getValue are strongly typed methods, enforcing non-null text, etc. 

Replacing it with a losely typed array does not feel good. 

I'm finishing to review format/code thing and i'll make a suggestion about the more global aspect :)